xds/internal/client/xds.go

```diff
@@ -631,21 +634,50 @@ func securityConfigFromCommonTLSContext(common *v3tlspb.CommonTlsContext) (*Secu
 	// those possible values:
 	//  - combined validation context:
 	//    - contains a default validation context which holds the list of
-	//      accepted SANs.
+	//      matchers for accepted SANs.
 	//    - contains certificate provider instance configuration
 	//  - certificate provider instance configuration
 	//    - in this case, we do not get a list of accepted SANs.
 	switch t := common.GetValidationContextType().(type) {
 	case *v3tlspb.CommonTlsContext_CombinedValidationContext:
 		combined := common.GetCombinedValidationContext()
+		var matchers []xds.StringMatcher
 		if def := combined.GetDefaultValidationContext(); def != nil {
-			for _, matcher := range def.GetMatchSubjectAltNames() {
-				// We only support exact matches for now.
-				if exact := matcher.GetExact(); exact != "" {
-					sc.AcceptedSANs = append(sc.AcceptedSANs, exact)
+			for _, m := range def.GetMatchSubjectAltNames() {
+				matcher := xds.StringMatcher{}
+				switch mt := m.GetMatchPattern().(type) {
+				case *v3matcherpb.StringMatcher_Exact:
+					matcher.ExactMatch = &mt.Exact
```

Comment:
Does empty `Exact` have any special meaning? I was wondering why only empty `Exact` is allowed, but not others like empty `Prefix`.